File: frontend/src/types/transaction.ts
export interface TransactionCreate {
  listing_id: number;
  message?: string;
}

export interface TransactionResponse {
  transaction_id: number;
  listing_id: number;
  buyer_id: number;
  seller_id: number;
  status: 'Pending' | 'Accepted' | 'Rejected' | 'Completed';
  offered_at: string;
  responded_at?: string;
  completed_at?: string;
  message?: string;
}

export interface TransactionUpdate {
  status: 'Pending' | 'Accepted' | 'Rejected' | 'Completed';
}

export interface TransactionWithListing extends TransactionResponse {
  listing?: {
    title: string;
    price: number;
    images?: string[];
  };
  buyer?: {
    first_name: string;
    last_name: string;
    email: string;
  };
  seller?: {
    first_name: string;
    last_name: string;
    email: string;
  };
} 

File: backend/app/schemas/transaction.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from enum import Enum

class TransactionStatus(str, Enum):
    PENDING = 'Pending'
    ACCEPTED = 'Accepted'
    REJECTED = 'Rejected'
    COMPLETED = 'Completed'

class TransactionCreate(BaseModel):
    listing_id: int
    message: Optional[str] = None

class TransactionResponse(BaseModel):
    transaction_id: int
    listing_id: int
    buyer_id: int
    seller_id: int
    status: TransactionStatus
    offered_at: datetime
    responded_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    message: Optional[str] = None
    
    model_config = {"from_attributes": True}

class TransactionUpdate(BaseModel):
    status: TransactionStatus
    model_config = {"from_attributes": True}

class ReviewCreate(BaseModel):
    transaction_id: int
    rating: float = Field(..., ge=1, le=5)
    comment: Optional[str] = None

class ReviewResponse(BaseModel):
    review_id: int
    transaction_id: int
    reviewer_id: int
    reviewee_id: int
    rating: float
    comment: Optional[str] = None
    created_at: datetime
    
    model_config = {"from_attributes": True}